compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs

```diff
@@ -118,13 +119,22 @@ data ShakeTestEnv = ShakeTestEnv
     , steVirtualResourcesNotes :: TVar (Map VirtualResource T.Text)
     }
 
+type TemplateName = String
+type ChoiceName = String
+
+data ExpectedChoiceAction
+    = Create TemplateName
+    | Exercise TemplateName ChoiceName deriving (Eq, Ord, Show)
+
 data ExpectedChoices = ExpectedChoices
     { _cName :: String
     , _consuming :: Bool
     } deriving (Eq, Ord, Show )
+
 data TemplateProp = TemplateProp
-    { _choices :: Set.Set ExpectedChoices
-    , _action :: Int
+    { _tplName :: T.Text
+    , _choices :: Set.Set ExpectedChoices
+    , _action :: Set.Set ExpectedChoiceAction
```

Comment:
The `_action` field should be moved into `ExpectedChoices`.

Reply:
Yup, thanks will move it